### PR TITLE
[XPU][Bugfix] Fix FP8 block-scaled GEMM for unaligned N

### PR DESCRIPTION
**Pull request by chaojun-zhang**
## Summary

Fix `XPUFp8BlockScaledMMKernel.apply_block_scaled_mm` to correctly derive
`block_size` from the weight scale tensor, and pad weight N-dimension when
it is not aligned to `block_size`. This fixes DeepSeek MTP model
initialization on XPU.

## Root Cause

The block_size was derived as `K // Bs.shape[0]`, but `Bs` has shape `[ceil(N/bs), ceil(K/bs)]`, so `Bs.shape[0]` is the N-group count, not K-group count. For `kv_a_proj_with_mqa` (N=576, K=2560, block_size=128):

- **Before**: `bs = 2560 // 5 = 512` → pads N to 1024 → `group_n = 1024/5 = 204` → oneDNN rejects
- **After**: `bs = 2560 // 20 = 128` → pads N to 640 → `group_n = 640/5 = 128` → works

## Changes

- **`vllm/model_executor/kernels/linear/scaled_mm/xpu.py`**:  Add N-dimension padding and output slicing for unaligned weights.
- **`tests/models/registry.py`**: Add `max_model_len=4096` for `DeepseekV3ForCausalLM`, `EagleDeepSeekMTPModel`, and `DeepSeekMTPModel` to prevent OOM on 24 GiB XPU devices.
- **`.buildkite/intel_jobs/misc_intel.yaml`**: Enable additional model initialization tests in Intel CI.

## Test
```bash
VLLM_XPU_FUSED_MOE_USE_REF=1 pytest -v -s \
  models/test_initialization.py::test_can_initialize_small_subset[DeepSeekMTPModel]

### CODE REVIEW

**Comment by jikunshang** [comment]
@Yejing-Lai  @zufangzhu please take a look.

File: tests/models/registry.py [hunk 1]
```diff
@@ -258,6 +258,7 @@ def check_available_online(
     "DeepseekV3ForCausalLM": _HfExamplesInfo(
         "deepseek-ai/DeepSeek-V3",
         trust_remote_code=True,
+        max_model_len=4096,
```
Comment (jikunshang) [inline tests/models/registry.py:259]: we should not add this.

Comment (chaojun-zhang) [reply tests/models/registry.py:259]: You’ll find that many model configs set max_model_len in this `registry.py`, and that value is dynamically loaded in the unit tests. without this overriding , this model can't be loaded .

File: vllm/model_executor/kernels/linear/scaled_mm/xpu.py [hunk 2]
```diff
@@ -204,13 +204,25 @@ def apply_block_scaled_mm(
         As: torch.Tensor,
         Bs: torch.Tensor,
     ) -> torch.Tensor:
+        # oneDNN block-quant requires N % block_size == 0.
```
Comment (jikunshang) [inline vllm/model_executor/kernels/linear/scaled_mm/xpu.py]: I feel all deepseek style block fp8 kernel require 128 aligned, not limit to xpu/onednn.

Comment (chaojun-zhang) [reply vllm/model_executor/kernels/linear/scaled_mm/xpu.py]: You are right,  updated

File: vllm/model_executor/kernels/linear/scaled_mm/xpu.py [hunk 3]
```diff
@@ -204,13 +204,26 @@ def apply_block_scaled_mm(
...
     ) -> torch.Tensor:
+        # Block FP8 format (DeepSeek-style) requires N % bs == 0 (block_size=128).
+        # Some model layers (e.g., DeepSeek MTP kv_a_proj_with_mqa with N=576) do not
+        # satisfy this condition, so we pad the dimension at runtime to avoid kernel
+        # out-of-bounds access.
+        N, K = B.shape
+        bs = K // Bs.shape[1] if Bs.shape[1] > 0 else 1
+        n_padded = ((N + bs - 1) // bs) * bs
+        if n_padded != N:
+            B = torch.nn.functional.pad(B, (0, 0, 0, n_padded - N))
```
Comment (Yejing-Lai) [inline vllm/model_executor/kernels/linear/scaled_mm/xpu.py]: Follow up [cuda path](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/kernels/linear/scaled_mm/cutlass.py#L312). I think we can add the unaligned N case on vllm-xpu-kernels, or add process_weights_after_loading on vllm side, it can avoid the weight pad cost on every forward.

**Comment by zufangzhu** [comment]
Why we need this pr? I remember you can't even load the block fp8 model weights when N is not 128 aligned in vllm. 
If only for UT, why not just skip them?
But if you insist on applying this fix, please make sure there is no any perf regression in block-fp8 model workload.

**Comment by chaojun-zhang** [comment]
> Why we need this pr? I remember you can't even load the block fp8 model weights when N is not 128 aligned in vllm. If only for UT, why not just skip them? But if you insist on applying this fix, please make sure there is no any perf regression in block-fp8 model workload.

Ok, will update accuray & perf result .

File: vllm/model_executor/kernels/linear/scaled_mm/xpu.py (hunk 3 above)
Comment (chaojun-zhang) [reply vllm/model_executor/kernels/linear/scaled_mm/xpu.py]: Updated and padding at load time instead of runtime ,please check again .
